src/components/Editor/index.js

```diff
@@ -160,16 +164,35 @@ class Editor extends Component {
     } else {
       this.undoTimestamp = 0
     }
+
+    if (evt.keyCode === 27 && !this.state.tabGuarded) {// Esc Key
+      this.setState({tabGuarded:true})
+    }else if(!(evt.shiftKey || evt.keyCode === 27 || evt.keyCode === 9) && this.state.tabGuarded){
```

Comment:
I just realised that we're being a bit silly here haha 😆 We can actually just set `tabGuarded` to false inside an `onChange` handler. So we create a new callback method, proxy it up and:

```js
onChange = evt => {
  if (this.props.onChange) {
    this.props.onChange(evt)
  }

  if (this.state.tabGuarded) {
    this.setState({ tabGuarded:false })
  }
}
```

That should work, right?